dask/array/creation.py

```diff
@@ -143,3 +144,36 @@ def arange(*args, **kwargs):
         elem_count += bs
 
     return Array(dsk, name, chunks, dtype=dtype)
+
+
+def diag(v):
+    """Construct a diagonal array, with ``v`` on the diagonal.
+
+    TODO: implement diagonal extraction, and the ``k``th diagonal keyword arg.
```

Comment:
I just put it in there because there was one in `linspace` already. I agree, and can remove it.